Mathlib/Condensed/Light/Functors.lean

```diff
@@ -0,0 +1,45 @@
+/-
+Copyright (c) 2023 Dagur Asgeirsson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Dagur Asgeirsson
+-/
+import Mathlib.CategoryTheory.Sites.Coherent.CoherentSheaves
+import Mathlib.Condensed.Light.Basic
+
+/-!
+# Functors from categories of topological spaces to light condensed sets
+
+This file defines the embedding of the test objects (light profinite sets) into light condensed
+sets.
+
+## Main definitions
+
+* `lightProfiniteToLightCondSet : LightProfinite.{u} ⥤ LightCondSet.{u}` 
+  is the yoneda presheaf functor.
+
+TODO (Dagur):
+
+* Add the functor `TopCat.{u} ⥤ LightCondSet.{u}`.
+
+* Define the functor `Type u ⥤ LightCondSet.{u}` which takes a set `X` to the presheaf given by
+  mapping a light profinite space `S` to `LocallyConstant S X`, along with the isomorphism with
+  the functor that goes through `TopCat.{u+1}`.
+
+-/
+
+universe u v
+
+open CategoryTheory Limits
+
+/-- The functor from `LightProfinite.{u}` to `LightCondSet.{u}` given by the Yoneda sheaf. -/
+def lightProfiniteToLightCondSet : LightProfinite.{u} ⥤ LightCondSet.{u} where
```

Comment:
I think it would be nice to define this as a particular case of the `yoneda` functor to sheaves of types for a subcanonical Grothendieck topology. Could you add the definition which is in draft form at https://github.com/leanprover-community/mathlib4/blob/jriou_localization/Mathlib/CategoryTheory/Sites/Canonical.lean#L253 and redefine `lightProfiniteToLightCondSet`.

Reply:
Thanks for the suggestion. It's added.

Reply:
Great! Could you also define a `FullyFaithful` structure for this and get `Full/Faithful` instances?